snap/local/hooks/cmd/configure/configure.go

```diff
@@ -0,0 +1,268 @@
+ */
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	hooks "github.com/canonical/edgex-snap-hooks/v2"
+)
+
+var cli *hooks.CtlCli = hooks.NewSnapCtl()
+
+const ( // iota is reset to 0
+	devVirtService = iota
+	kuiperService
+	secProxyService
+	secStoreService
+	otherService
+)
+
+const (
+	ON  = "on"
+	OFF = "off"
+)
+
+// handleSingleService starts or stops a service based on
+// the given state (ON|OFF). It also ensures that the top
+// level service configuration option is set accordingly.
+func handleSingleService(name, state string) error {
+
+	switch state {
+	case OFF:
+		hooks.Debug("edgexfoundry:configure: state: off")
+		if err := cli.Stop(name, true); err != nil {
+			return err
+		}
+		if err := cli.SetConfig(name, OFF); err != nil {
+			return err
+		}
+	case ON:
+		hooks.Debug("edgexfoundry:configure: state: on")
+		if err := cli.Start(name, true); err != nil {
+			return err
+		}
+		if err := cli.SetConfig(name, ON); err != nil {
+			return err
+		}
+	case "":
+		hooks.Debug("edgexfoundry:configure: state: ''")
+	default:
+		return fmt.Errorf("edgexfoundry:configure: invalid state %s for service: %s", state, name)
+	}
+
+	return nil
+}
+
+func serviceType(name string) int {
+	switch name {
+	case hooks.ServiceDevVirt:
+		return devVirtService
+	case hooks.ServiceKuiper:
+		return kuiperService
+	case hooks.ServiceProxy:
+		return secProxyService
+	case hooks.ServiceSecStore:
+		return secStoreService
+	default:
+		return otherService
+	}
+}
+
+// handleAllServices iterates through all of the services in the
+// edgexfoundry snap and:
+//
+// - queries the config option associated with the service state (on|off)
+// - queries the environment configuration for the service (env.<service-name>)
+//   - if env configuration for the service exists, use it to write a
+//     service-specific .env file to the service config dir in $SNAP_DATA
+// - start/stop any tightly couple services (e.g. if the secret-store
+//   is disabled, the proxy also has to come down) if required
+// - start/stop the service itself if required
+//
+// NOTE - at this time, this function does *not* restart a service based
+// on env configuration changes. If changes are made after a service has
+// been started, the service must be restarted manually.
+//
+func handleAllServices() error {
+
+	// grab and log the current service configuration
+	for _, s := range hooks.Services {
+		var envJSON string
+
+		status, err := cli.Config(s)
+		if err != nil {
+			return err
+		}
+
+		hooks.Debug(fmt.Sprintf("edgexfoundry:configure: service: %s status: %s", s, status))
+
+		serviceCfg := hooks.EnvConfig + "." + s
+		envJSON, err = cli.Config(serviceCfg)
+		if err != nil {
+			return fmt.Errorf("edgexfoundry:configure failed to read service %s configuration - %v", s, err)
+		}
+
+		if envJSON != "" {
+			hooks.Debug(fmt.Sprintf("edgexfoundry:configure: service: %s envJSON: %s", s, envJSON))
+			if err := hooks.HandleEdgeXConfig(s, envJSON, nil); err != nil {
```

Comment:
See the comment in `snap/local/hooks/cmd/install/install.go` re: the `extraConf` parameter. Because you don't pass it here, if someone were to set their own value for `add-known-secrets` or `add-registry-acl-roles`, the configure hook wouldn't be able to process them.

Reply:
yes, which will be resolved by adding this to edgex-snap-hooks.